### PR TITLE
#1425 sfsection remove underline modifier

### PR DESCRIPTION
**Pull request by JonathanDn**
# Related issue
Closes https://github.com/DivanteLtd/storefront-ui/issues/1425

# Scope of work
Deprecated usage of `sf-heading--underline` class by -
removing all its logic class from files:
SfSection.vue , SfSection.stories.js, SfHeading.stories.js

# Screenshots of visual changes

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed
- [x] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/atoms/SfHeading/SfHeading.stories.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ storiesOf("Atoms|Heading", module)
         default: options(
           "CSS modifiers",
           {
-            "sf-heading--underline": "sf-heading--underline",
```
Comment (justyna-13) [inline packages/vue/src/components/atoms/SfHeading/SfHeading.stories.js:18]: these changes are connected with SfHeading, which is not a part of your issue. Please, don't delete modifiers in SfHeading :slightly_smiling_face:

File: packages/vue/src/components/molecules/SfSection/SfSection.vue [hunk 2]
```diff
@@ -50,15 +49,9 @@ export default {
     },
   },
   data() {
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfSection/SfSection.vue:51]: if data is empty, you can delete it  :)

File: packages/vue/src/components/molecules/SfSection/SfSection.vue [hunk 3]
```diff
@@ -50,15 +49,9 @@ export default {
...
-      hasUnderlinedModifier: false,
-    };
-  },
-  mounted: function () {
-    this.hasUnderlinedModifier = this.$el.classList.contains(
-      "sf-section--underline"
-    );
+    return {};
   },
+  mounted: function () {},
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfSection/SfSection.vue:54]: same as above :) mounted is empty, so you can delete it

**Review by justyna-13 — CHANGES_REQUESTED** [review]
@JonathanDn thank you for your PR :+1:  please, take a look at the comments regarding your changes

**Comment by JonathanDn** [comment]
> @JonathanDn thank you for your PR  please, take a look at the comments regarding your changes

All fixed according to the comments.

**Comment by JonathanDn** [comment]
@justyna-13 It seems Travis Build is stuck on `pending` for a while, maybe it should be reset?

File: packages/vue/src/components/molecules/SfSection/SfSection.stories.js [hunk 4]
```diff
@@ -14,10 +14,6 @@ storiesOf("Molecules|Section", module)
       customClass: {
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfSection/SfSection.stories.js:14]: you can also delete both customClasses in this file :) we don't need them anymore

File: packages/vue/src/components/molecules/SfSection/SfSection.vue [hunk 5]
```diff
@@ -48,17 +47,7 @@ export default {
...
-    return {
-      hasUnderlinedModifier: false,
-    };
-  },
-  mounted: function () {
-    this.hasUnderlinedModifier = this.$el.classList.contains(
-      "sf-section--underline"
-    );
-  },
+  }
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfSection/SfSection.vue:50]: add a comma after curly brackets :)

**Review by justyna-13 — COMMENTED** [review]
Looks great! Just a few fixes :slightly_smiling_face:

File: packages/vue/src/components/molecules/SfSection/SfSection.stories.js (hunk 4 above)
Comment (JonathanDn) [reply packages/vue/src/components/molecules/SfSection/SfSection.stories.js]: done

Comment (JonathanDn) [reply packages/vue/src/components/molecules/SfSection/SfSection.stories.js]: > Looks great! Just a few fixes

Fixed the 2 last comments

**Review by JonathanDn — COMMENTED** [review]
> Looks great! Just a few fixes
Fixed the last 2 issues.
- added `,` 
- removed the 2 `customClass` definitions completely from the file `SfSection.stories.js`

**Review by justyna-13 — APPROVED** [review]
:+1: great @JonathanDn!

**Review by AdamPawlinski — APPROVED** [review]

**Comment by AdamPawlinski** [comment]
Good work @JonathanDn, thanks for taking care of it :) And thank you @justyna-13 for your assistance.

**Comment by JonathanDn** [comment]
@AdamPawlinski My pleasure :)

**Pull request merged**